### ISSUE TITLE
Change consumer subscription API

### ISSUE THREAD

**Issue by dasch** [zendesk/ruby-kafka#207]
Currently, the API looks like this:

```ruby
consumer.subscribe("x", default_offset: :earliest)
consumer.subscribe("y", default_offset: :latest)
```

I feel that it's a somewhat confusing API. From the point of view of the user, it's really a boolean option of whether to _start from the beginning_ or not. As such, the API could look like this:

```ruby
consumer.subscribe("x", start_from_beginning: true)
consumer.subscribe("y", start_from_beginning: false)
```

**Comment by dasch** [comment]
@osheroff does this make sense?

**Comment by dansimpson** [comment]
Overall, it makes sense, but having some language to convey that the flag only takes effect when no committed offsets are present helps.  I ran into that question when first using the consumer API.

**Comment by dasch** [comment]
@dansimpson is there some wording that would have helped?

**Comment by dansimpson** [comment]
@dasch I think adding another sentence to the documentation for the subscribe method works:

```
@param default_offset [Symbol] whether to start from the beginning or the end of the topic's partitions.  This parameter only affects consumers which have not committed offsets for the topic.
```

I think this works with start_from_beginning as well.

**Comment by dasch** [comment]
Closed by #209.
